syftlib/src/main/java/org/openmined/syft/device/repositories/NetworkStatusRepository.kt

```diff
@@ -0,0 +1,143 @@
+package org.openmined.syft.device.repositories
+
+import android.accounts.NetworkErrorException
+import android.content.Context
+import android.net.ConnectivityManager
+import android.util.Log
+import io.reactivex.Completable
+import io.reactivex.Single
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.MultipartBody
+import okhttp3.RequestBody.Companion.asRequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.ResponseBody
+import org.openmined.syft.device.models.NetworkStateModel
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.utilities.FileWriter
+import org.openmined.syft.utilities.MB
+import org.openmined.syft.utilities.readNBuffers
+import java.util.Random
+
+private const val TAG = "NetworkStateRepository"
+
+private const val SPEED_BUFFER_WINDOW = 20
+private const val SPEED_MULTIPLICATION_FACTOR = 10
+private const val MAX_SPEED_TESTING_BYTES = MB * 8
+
+@ExperimentalUnsignedTypes
+class NetworkStatusRepository(private val configuration: SyftConfiguration)  {
```

Comment:
I would have put `NetworkStatusRepository` and `SleepStatusRepository` in a separate Android module and leave the library to define the contract.
But I understand it's easier like this. We can refactor later if we think we need to.

Reply:
I was thinking about separating them but the syft worker would still need to call out these test before running the cycle. The syft worker will be needing device monitor so I am not sure if syft.kt could remain independent of android code